src/api.rs

```diff
@@ -203,8 +201,8 @@ unsafe fn join_inject<A,B,RA,RB>(oper_a: A,
 
     thread_pool::get_registry().inject(&[job_a.as_job_ref(), job_b.as_job_ref()]);
 
-    job_a.latch.wait();
-    job_b.latch.wait();
+    job_a.wait();
```

Comment:
It's mostly cleanup, but it's needed since the `wait_all` method needs to call `wait()` on the tasks.